contrib/babelfishpg_tsql/src/dbcmds.c

```diff
@@ -406,20 +386,61 @@ create_bbf_db_internal(const char *dbname, List *options, const char *owner, int
 	bool 			is_set_userid = false;
 	Oid 			save_userid;
 	const char	*old_createrole_self_grant;
+	ListCell	*option;
+	const char* database_collation_name = NULL;
 
-	/* TODO: Extract options */
-
-	tuple = SearchSysCache1(COLLOID, ObjectIdGetDatum(tsql_get_server_collation_oid_internal(false)));
-	if (!HeapTupleIsValid(tuple))
+	/* Check options */
+	foreach(option, options)
 	{
-		const char *server_collation_name = GetConfigOption("babelfishpg_tsql.server_collation_name", false, false);
+		DefElem    *defel = (DefElem *) lfirst(option);
 
-		ereport(ERROR,
-				(errcode(ERRCODE_UNDEFINED_OBJECT),
-				 errmsg("OID corresponding to collation \"%s\" does not exist", server_collation_name)));
+		if (strcmp(defel->defname, "collate") == 0)
+		{
+			database_collation_name = pstrdup(defGetString(defel));
+
+			if (tsql_find_collation_internal(database_collation_name) == NOT_FOUND)
+			{
+				database_collation_name = tsql_translate_tsql_collation_to_bbf_collation(database_collation_name);
+				if (tsql_find_collation_internal(database_collation_name) == NOT_FOUND)
+				{
+					ereport(ERROR,
+						(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+						 errmsg("Invalid collation \"%s\"", defGetString(defel)),
+							parser_errposition(pstate, defel->location)));
+				}
+			}
```

Comment:
Can we simply this logic a bit? 1st, lets make `translate_tsql_collation_to_bbf_collation` to return original collation instead of NULL in case translated collation is not found. And then we can directly, do something like following,
```
database_collation_name = tsql_translate_tsql_collation_to_bbf_collation(defGetString(defel))
if (tsql_find_collation_internal(database_collation_name) == NOT_FOUND)
{
// error
}
```